cmd/kpod/launch.go

```diff
@@ -1,16 +1,592 @@
+		Annotations: make(map[string]string),
+		// TODO populate cgroup and security configuration
+		Linux: &linuxSandboxConfig,
+	}
+
+	return &config, nil
+}
+
+func makeContainerCreateRequest(cliConfig *launchConfig, securityConfig *pb.LinuxContainerSecurityContext, sandboxID string, sandboxConfig *pb.PodSandboxConfig) (*pb.CreateContainerRequest, error) {
+	metadata := pb.ContainerMetadata{
+		Name:    cliConfig.containerName,
+		Attempt: 0,
+	}
+
+	linuxConfig := pb.LinuxContainerConfig{
+		Resources:       cliConfig.limits,
+		SecurityContext: securityConfig,
+	}
+
+	config := pb.ContainerConfig{
+		Metadata: &metadata,
+		Image:    &pb.ImageSpec{Image: cliConfig.image},
+		Command:  []string{cliConfig.command},
+		Args:     []string{},
+		// TODO: allow passing this via CLI (if worthwhile)
+		WorkingDir: "",
+		Envs:       cliConfig.env,
+		Mounts:     cliConfig.mount,
+		Devices:    cliConfig.devices,
+		Labels:     make(map[string]string),
+		// TODO allow setting annotations
+		Annotations: nil,
+		// TODO Logging
+		LogPath:   "",
+		Stdin:     cliConfig.stdin,
+		StdinOnce: cliConfig.stdinOnce,
+		Tty:       cliConfig.tty,
+		Linux:     &linuxConfig,
+	}
+
+	if cliConfig.args != nil {
+		config.Args = *cliConfig.args
+	}
+
+	if cliConfig.labels != nil {
+		config.Labels = *cliConfig.labels
+	}
+
+	req := pb.CreateContainerRequest{
+		PodSandboxId:  sandboxID,
+		Config:        &config,
+		SandboxConfig: sandboxConfig,
+	}
+
+	return &req, nil
+}
+
+// TODO: Capabilities, SELinux
+func generateLinuxSecurityConfigs(cliConfig *launchConfig) (*pb.LinuxSandboxSecurityContext, *pb.LinuxContainerSecurityContext, error) {
+	linuxNamespaceOption := pb.NamespaceOption{
+		HostNetwork: cliConfig.hostNet,
+		HostPid:     cliConfig.hostPid,
+		HostIpc:     cliConfig.hostIpc,
+	}
+
+	runAsUser := pb.Int64Value{
+		Value: cliConfig.runAsUser,
+	}
+
+	sandboxConfig := pb.LinuxSandboxSecurityContext{
+		NamespaceOptions:   &linuxNamespaceOption,
+		SelinuxOptions:     nil,
+		RunAsUser:          &runAsUser, // Probably not strictly necessary to set this here if we're doing it below
+		ReadonlyRootfs:     cliConfig.readOnlyRoot,
+		SupplementalGroups: cliConfig.additionalGroups,
+		Privileged:         cliConfig.privileged,
+	}
+
+	// TODO: We can't get group independently of user. SupplementalGroups helps alleviate this, but it's desirable.
+	// Adding this would require changes to the server, so it's not really worth it for now
+	containerConfig := pb.LinuxContainerSecurityContext{
+		Capabilities:       &pb.Capability{},
+		Privileged:         cliConfig.privileged,
+		NamespaceOptions:   &linuxNamespaceOption,
+		SelinuxOptions:     nil,
+		RunAsUser:          &runAsUser,
+		RunAsUsername:      "",
+		ReadonlyRootfs:     cliConfig.readOnlyRoot,
+		SupplementalGroups: cliConfig.additionalGroups,
+	}
+
+	return &sandboxConfig, &containerConfig, nil
+}
+
+// Generate and hex-encode 128-bit random ID
+func getRandomID() (string, error) {
```

Comment:
We use github.com/docker/docker/pkg/stringid.GenerateNonCryptoID() for this elsewhere in the server package.